tests/baselines/reference/dependentReturnType11(strict=true).errors.txt

```diff
@@ -0,0 +1,19 @@
+dependentReturnType11.ts(7,5): error TS2322: Type 'T' is not assignable to type 'T extends Nullable ? Nullable : never'.
+  Type 'null' is not assignable to type 'T extends Nullable ? Nullable : never'.
+
+
+==== dependentReturnType11.ts (1 errors) ====
+    type Nullable = null | undefined;
+    
+    function test1<T extends null>(
+      x: T,
+    ): T extends Nullable ? Nullable : never {
+      if (x == undefined) {
+        return x;

```

Comment:
in theory, this shouldn't report an error but this code is just awfully non-sensical, I only have added it as extra precautions against potential crashes (that didn't happen - but somewhat only because `some` helper is OK with `undefined` argument)